somecode.cpp

```diff
@@ -0,0 +1,32 @@
+#include <stdio.h>
+#include <string.h>
+
+int main()
+{
+   char src[40];
+   char dest[100];
+   char dest2[100];
+
+   printf("Enter your name");
+   gets(src); 
+
+   // Copy name from one place to another
+   strcpy(dest, src); 
+ 
+   int a = 10;
+   int b = 20;
+   int c = a+b;
+   printf("Result is %d",&c);
+ 
+   strncpy(dest,src, strlen(dest)); 
+   
+   for (int i=0; i < 10; i++)
+   {
+      strncat(dest, "this is bad", strlen(dest)); 
```

Comment:
[Banned C function detected (strncat)](https://github.com/Microsoft/DevSkim/blob/master/guidance/DS108330.md#DS108330)
strncat adds the null terminator at character 'n + 1', rather than at the nth character. this frequently leads to the null terminator being added in the memory adjacent to the destination buffer, rather than in the destination buffer.

*Fix Guidance:* Use strcat_s or strlcat if possible. 


Suggestions:
[1] strcat_s(dest,  strlen(dest),  "this is bad");
[2] strlcat(dest,  "this is bad",  strlen(dest));